kotlin-options/src/main/kotlin/com/petertackage/kotlinoptions/Option.kt

```diff
@@ -3,6 +3,14 @@ package com.petertackage.kotlinoptions
 fun <T : Any> optionOf(value: T?): Option<T> =
     if (value == null) None else Some(value)
 
+inline fun <T : Any> tryAsOption(function: () -> T?): Option<T> {
```

Comment:
try is an expression so it could be:
```Kotlin
inline fun <T : Any> tryAsOption(function: () -> T?): Option<T> {
    return try {
        optionOf(function())
    } catch (e: Exception) {
        None
    }
}
```